### ISSUE TITLE
Features for python hydra client

### ISSUE THREAD

**Issue by ashwani99** [HTTP-APIs/python-hydra-agent#6]
Currently the Hydra client lacks the following:
- support for classes like `hydra:Error`, `hydra:PartialCollectionView`, hydra:Link` etc
- linked data fragments
- extensive documentation for classes and functions (docstrings can be used)
- support for python 3 and type annotations

**Comment by Mec-iS** [comment]
> support for classes like hydra:Error, hydra:PartialCollectionView, hydra:Link etc

Absolutely. The client has to be updated with most recent specs and use cases as for [Heracles](https://github.com/HydraCG/Heracles.ts)

> linked data fragments

As in #3

> extensive documentation for classes and functions (docstrings)

Absolutely yes.

> support for python 3(with backward compatibility) and type annotations

Absolutely yes.

**Comment by ashwani99** [comment]
@Mec-iS Thanks. I am working on a draft proposal for this. I am planning to club it with the CLI and Falcon port. Can you please help me if this approach is fine or is it too much to work?

**Comment by Mec-iS** [comment]
Share the draft and you will receive feedback in your google doc.
